dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt

```diff
@@ -0,0 +1,353 @@
+    val pkJson = publicKeyJWK.toJSONString()
+
+    // 2. Canonicalize the JWK encoded public key using the implementation’s JSON_CANONICALIZATION_SCHEME.
+    val canonicalized = JsonCanonicalizer(pkJson).encodedUTF8
+
+    // 3. Use the implementation’s HASH_PROTOCOL to Multihash the canonicalized public key to generate the REVEAL_VALUE,
+    val intermediate = Multihash.sum(Multicodec.SHA2_256, canonicalized).get()?.digest!!
+
+    // then Multihash the resulting Multihash value again using the implementation’s HASH_PROTOCOL to produce
+    // the public key commitment.
+    val hashOfHash = Multihash.sum(Multicodec.SHA2_256, intermediate).get()?.bytes()
+    return Base64URL.encode(hashOfHash).toString()
+  }
+
+  /**
+   * Default companion object for creating a [DidIonManager] with a default configuration.
+   */
+  public companion object Default : DidIonManager(DidIonConfiguration())
+}
+
+/**
+ * Represents an HTTP response where the status code is outside the range considered success.
+ */
+public class InvalidStatusException(s: String) : Exception(s)
+
+/** Wraps an exception during resolution where the [DidResolutionMetadata.error] is not empty. */
+public class ResolutionException(s: String) : Exception(s)
+
+/**
+ * Container for the key aliases for an ION did.
+ */
+public data class KeyAliases(
+  public val updateKeyAlias: String,
+  public val verificationKeyAlias: String,
+  public val recoveryKeyAlias: String)
+
+/**
+ * Options available when creating an ion did.
+ *
+ * @param verificationPublicKey When provided, will be used as the verification key in the DID document.
+ * @param updatePublicJWK When provided, will be used to create the update key commitment.
+ * @param recoveryPublicJWK When provided, will be used to create the recovery key commitment.
+ * @param verificationMethodId When provided, will be used as the verification method id. Cannot be over 50 chars and
+ * must only use characters from the Base64URL character set.
+ */
+public class CreateDidIonOptions(
+  public val verificationPublicKey: PublicKey? = null,
+  public val updatePublicJWK: JWK? = null,
+  public val recoveryPublicJWK: JWK? = null,
+  public val verificationMethodId: String? = null,
+) : CreateDidOptions
+
+/**
+ * Metadata related to the creation of a DID (Decentralized Identifier) on the Sidetree protocol.
+ *
+ * @property createOperation The Sidetree create operation used to create the DID.
+ * @property shortFormDid The short-form DID representing the DID created.
+ * @property longFormDid The long-form DID representing the DID created.
+ * @property operationsResponseBody The response body received after submitting the create operation.
+ */
+public data class IonCreationMetadata(
+  public val createOperation: SidetreeCreateOperation,
+  public val shortFormDid: String,
+  public val longFormDid: String,
+  public val operationsResponseBody: String,
+  public val keyAliases: KeyAliases,
+) : CreationMetadata
```

Comment:
what's the deciding factor as to whether something is considered a model or not?